### PR TITLE
Updated Connect create cluster command to include offsets

### PR DESCRIPTION
**Pull request by nmala01**
Release Notes
-------------
New Features
- Support passing offsets to `confluent connect cluster create` as part of the configuration file

What
----
Updated Connect create cluster command to include offsets

References
----------

Test & Review
-------------
Updated unit tests to include offsets

### CODE REVIEW

File: internal/connect/command_cluster_create.go [hunk 1]
```diff
@@ -25,13 +25,44 @@ func (c *clusterCommand) newCreateCommand() *cobra.Command {
 		RunE:        c.create,
 		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
 		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "The configuration file contains configs including offsets for the connector.",
+				Code: `{
```
Comment (naveenmall11) [inline internal/connect/command_cluster_create.go:30]: Code inside example is different from other examples, is it expected here?

File: test/fixtures/input/connect/config.yaml [hunk 2]
```diff
@@ -9,5 +9,21 @@
   "tasks.max": "1",
   "time.interval": "HOURLY",
   "topics": "apples",
-  "connector.class": "AzureBlobSink"
+  "connector.class": "AzureBlobSink",
```
Comment (naveenmall11) [inline test/fixtures/input/connect/config.yaml:12]: This seems to be input file for CreateORUpdateConnector and offsets field is not supported in this API. Can you please check where is this file being used?

File: internal/connect/command_cluster_create.go (hunk 1 above)
Comment (nmala01) [reply internal/connect/command_cluster_create.go:30]: Yes, kept in consistent with update offset.

File: test/fixtures/input/connect/config.yaml (hunk 2 above)
Comment (nmala01) [reply test/fixtures/input/connect/config.yaml:12]: for CreateOrUpdate, we are not passing the offset.

**Review by naveenmall11 — APPROVED** [review]
LGTM!!

**Comment by brianstrauch** [comment]
This looks like a user-facing change, so can you fill in the release notes section?

File: internal/connect/command_cluster_create.go [hunk 3]
```diff
@@ -25,6 +25,36 @@ func (c *clusterCommand) newCreateCommand() *cobra.Command {
 		RunE:        c.create,
 		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
 		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "The configuration file contains configs including offsets for the connector.",
```
Comment (brianstrauch) [inline internal/connect/command_cluster_create.go:29]:
```suggestion
				Text: "Create a configuration file with connector offsets.",
```

File: internal/connect/command_cluster_create.go [hunk 4]
```diff
@@ -25,6 +25,36 @@ func (c *clusterCommand) newCreateCommand() *cobra.Command {
...
+				Code: `{
+  "name": "MyGcsLogsBucketConnector",
```
Comment (brianstrauch) [inline internal/connect/command_cluster_create.go:31]: What are GCS logs? Can we update the text of this example to explain what they are?

File: internal/connect/utils.go [hunk 5]
```diff
@@ -10,15 +10,15 @@ import (
 	"github.com/confluentinc/cli/v3/pkg/errors"
 )
 
-func getConfig(cmd *cobra.Command) (*map[string]string, error) {
+func getConfig(cmd *cobra.Command, isUpdate bool) (*map[string]string, *[]map[string]any, error) {
```
Comment (brianstrauch) [inline internal/connect/utils.go:13]: It looks like this function is doing two separate things. Can we split this into two functions: `getConfig()` and `getOffsets()`?

File: internal/connect/utils.go [hunk 6]
```diff
@@ -30,51 +30,63 @@ func getConfig(cmd *cobra.Command) (*map[string]string, error) {
...
 				kvPairs[configKey] = value
 			}
+		} else if key == "offsets" {
+			if isUpdate {
+				return nil, nil, fmt.Errorf(`offsets are not allowed in configuration file for update`)
+			}
+			var request *[]map[string]any
+			valBytes, err := json.Marshal(val)
+			if err != nil {
+				return nil, nil, fmt.Errorf(`error while marshallling offsets :- value for the configuration "offsets" is malformed`)
```
Comment (brianstrauch) [inline internal/connect/utils.go:79]: What does ":-" mean?

File: test/fixtures/output/connect/cluster/update-error.golden [hunk 7]
```diff
@@ -0,0 +1 @@
+Error: unable to read configuration file "test/fixtures/input/connect/update-config-malformed.json": offsets are not allowed in configuration file for update
```
Comment (brianstrauch) [inline test/fixtures/output/connect/cluster/update-error.golden:1]:
```suggestion
Error: unable to read configuration file "test/fixtures/input/connect/update-config-malformed.json": offsets are not allowed in updated configuration file
```

File: internal/connect/utils.go [hunk 8]
```diff
@@ -30,51 +30,63 @@ func getConfig(cmd *cobra.Command) (*map[string]string, error) {
...
+			}
+			if err := json.Unmarshal(valBytes, &request); err != nil {
+				return nil, nil, fmt.Errorf(`error while unmarshallling offsets :- value for the configuration "offsets" is malformed`)
```
Comment (sgagniere) [inline internal/connect/utils.go:82]:
```suggestion
				return nil, nil, fmt.Errorf(`error while marshalling offsets :- value for the configuration "offsets" is malformed`)
			}
			if err := json.Unmarshal(valBytes, &request); err != nil {
				return nil, nil, fmt.Errorf(`error while unmarshalling offsets :- value for the configuration "offsets" is malformed`)
```

File: internal/connect/command_cluster_create.go (hunk 4 above)
Comment (nmala01) [reply internal/connect/command_cluster_create.go:31]: Its just a example of connector type.  Doesnt have any specific relation here.

File: internal/connect/command_cluster_create.go (hunk 3 above)
Comment (nmala01) [reply internal/connect/command_cluster_create.go:29]: Not sure if `Create a configuration file with connector offsets.`  makes sense, we are inputting config file which contains user configs and offsets.

File: internal/connect/utils.go (hunk 5 above)
Comment (nmala01) [reply internal/connect/utils.go:13]: both is read from same file, i can rename the method.

File: test/fixtures/output/connect/cluster/update-error.golden (hunk 7 above)
Comment (nmala01) [reply test/fixtures/output/connect/cluster/update-error.golden:1]: May be  i can do `offsets are not allowed in configuration file for update command`

File: internal/connect/command_cluster_create.go (hunk 3 above)
Comment (nmala01) [reply internal/connect/command_cluster_create.go:29]: ok

File: test/fixtures/output/connect/cluster/update-error.golden (hunk 7 above)
Comment (brianstrauch) [reply test/fixtures/output/connect/cluster/update-error.golden:1]: > offsets are not allowed in configuration file for `confluent connect cluster update`

File: internal/connect/utils.go (hunk 5 above)
Comment (nmala01) [reply internal/connect/utils.go]: Done

File: internal/connect/utils.go (hunk 8 above)
Comment (nmala01) [reply internal/connect/utils.go]: Done

File: internal/connect/utils.go (hunk 6 above)
Comment (nmala01) [reply internal/connect/utils.go]: removed

File: internal/connect/utils.go [hunk 9]
```diff
@@ -10,15 +10,15 @@ import (
 	"github.com/confluentinc/cli/v3/pkg/errors"
 )
 
-func getConfig(cmd *cobra.Command) (*map[string]string, error) {
+func getConfigAndOffsets(cmd *cobra.Command, isUpdate bool) (*map[string]string, *[]map[string]any, error) {
```
Comment (brianstrauch) [inline internal/connect/utils.go:13]: Maps and slices are reference types, so they are always passed by reference. You don't need a pointer.
```suggestion
func getConfigAndOffsets(cmd *cobra.Command, isUpdate bool) (map[string]string, []map[string]any, error) {
```

**Review by sgagniere — APPROVED** [review]

**Review by linouk23 — APPROVED** [review]

**Pull request merged**